include/ruby/internal/core/robject.h

```diff
@@ -95,6 +97,14 @@ struct RObject {
     /** Basic part, including flags and class. */
     struct RBasic basic;
 
+#if USE_RVARGC
+    /**
+    * Number of instance variables.  This is per object; objects might
+    * differ in this field even if they have the identical classes.
+    */
+    uint32_t numiv;
+#endif
```

Comment:
I guess this is a matter of personal taste, but I find this code hard to parse. It might be more readable to have an `#if USE_RVARGC` that wraps the entire `RObject` struct, and two completely different struct definitions for the case with and without VWA.

But, IIUC, what you're doing here is making it so objects always have a `numiv` field, even when they are embedded.

Reply:
The goal is to remove all non-VWA code in the future, so it's probably not here to stay for very long.